### PR TITLE
tests: create table from source: miscellaneous test extensions

### PR DESCRIPTION
**Pull request by nrainer-materialize**
Part of https://github.com/MaterializeInc/materialize/issues/29510.

### CODE REVIEW

**Review by ParkMyCar — APPROVED** [review]
I would wait for @rjobanp for the final approval, but LGTM

**Review by rjobanp — APPROVED** [review]

**Review by def- — APPROVED** [review]

**Pull request merged**
